hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusDeleteTopicHandler.java

```diff
@@ -56,24 +57,18 @@ public ConsensusDeleteTopicHandler() {
      * @param topicStore the {@link ReadableTopicStore} to use to resolve topic metadata
      * @throws NullPointerException if any of the arguments are {@code null}
      */
-    public void preHandle(@NonNull final PreHandleContext context, @NonNull ReadableTopicStore topicStore) {
+    public void preHandle(@NonNull final PreHandleContext context, @NonNull ReadableTopicStore topicStore)
+            throws PreCheckException {
         requireNonNull(context);
         requireNonNull(topicStore);
 
-        final var op = context.getTxn().consensusDeleteTopicOrThrow();
-        final var topicMeta = topicStore.getTopicMetadata(op.topicIDOrElse(TopicID.DEFAULT));
-        if (topicMeta.failed()) {
-            context.status(ResponseCodeEnum.INVALID_TOPIC_ID);
-            return;
-        }
-
-        final var adminKey = topicMeta.metadata().adminKey();
-        if (adminKey.isEmpty()) {
-            context.status(ResponseCodeEnum.UNAUTHORIZED);
-            return;
-        }
-
-        context.addToReqNonPayerKeys(adminKey.get());
+        final var op = context.body().consensusDeleteTopicOrThrow();
+        // The topic ID must be present on the transaction and the topic must exist.
+        final var topic = topicStore.getTopicMetadata(op.topicID());
+        mustExist(topic, INVALID_TOPIC_ID);
```

Comment:
Probably `validateExistsOrThrow`  instead of `mustExist` ?

Reply:
🤔

Reply:
I don't mind it other than being long-winded.